src/tests/app.test.ts

```diff
@@ -1,32 +1,75 @@
 import 'jest';
 import supertest from 'supertest';
+import logger from '../logger';
 
-/* eslint-disable @typescript-eslint/no-var-requires, global-require */
+const loggerErrorSpy = jest.spyOn(logger, 'error').mockImplementation();
+const loggerInfoSpy = jest.spyOn(logger, 'info').mockImplementation();
+
+// eslint-disable-next-line import/first
+import { app, initSlack } from '../app';
+
+const processExitSpy = jest.spyOn(process, 'exit').mockImplementation();
+const mockSlackAuth = jest.fn();
+jest.mock('@slack/web-api', () => ({
+  // eslint-disable-next-line object-shorthand, func-names, @typescript-eslint/explicit-function-return-type, space-before-function-paren
+  WebClient: function() {
+    return {
+      auth: {
+        test: mockSlackAuth,
+      },
+    };
+  },
+}));
 
 describe('app', () => {
   beforeEach(() => {
-    process.env.SLACK_SIGNING_SECRET = 'A JUNK TOKEN';
-    process.env.SLACK_BOT_TOKEN = 'ANOTHER JUNK TOKEN';
+    Object.defineProperty(process.env, 'NODE_ENV', { value: 'test' });
+    jest.resetModules();
+    jest.resetAllMocks();
+    mockSlackAuth.mockRejectedValue('Invalid Auth');
   });
 
   it('responds successfully to GET /', (done) => {
-    const { app } = require('../app');
     supertest(app)
       .get('/')
       .expect(200, done);
   });
 
-  it("fails to start when Slack tokens are not provided and env isn't test", () => {
-    delete process.env.SLACK_BOT_TOKEN;
+  it("fails to initialize Slack when Slack tokens are not provided and env isn't test and exits", async () => {
     Object.defineProperty(process.env, 'NODE_ENV', { value: 'development' });
-    expect(require('../app').app).toThrow();
-    // Reset the env variable to 'test'
+    jest.resetModules();
+    await initSlack();
+    expect(processExitSpy).toBeCalledTimes(1);
+  });
+
+  it('fails to initialize Slack, logs an error, but continues in test env', async () => {
```

Comment:
To match previous change request.

```suggestion
  it('will not exit the process when Slack tokens are not provided and NODE_ENV === "test", but will log an error', async () => {
```

What's the purpose of it logging an error in the test environment if it's mocked? I would think it would be better for the error to be logged when it quits, otherwise it's not providing much use. And if it is currently being logged when it quits, then I would just move
```typescript
expect(loggerErrorSpy).toBeCalledTimes(1);
```
up into the above test so that we're testing what we want to happen.

Reply:
The purpose of this is to make sure tests still function correctly when they aren't specifying Slack credentials. I agree the logger serves no purpose; I'll remove it